.github/workflows/GHA-Unit-Tests.yaml

```diff
@@ -89,3 +89,89 @@ jobs:
           # that were leftover in unrelated modules for functional and instrumentation tests.
           path: |
             **/build/reports/tests/*
+
+      - name: Checkout GhPages branch
+        if: ${{ failure() }}
+        uses: actions/checkout@v3
+        with:
+          ref: 'gh-pages'
+          path: 'gh-pages'
+
+      - name: Configure Git
+        if: ${{ failure() }}
+        run: |
+          git config --global user.name "GitHub Actions Bot"
+          git config --global user.email "actions@users.noreply.github.com"
+
+      - name: Download test reports Java ${{ matrix.java-version }}
+        if: ${{ failure() }}
+        uses: actions/download-artifact@v3
+        with:
+          name: unit-tests-results-java-${{ matrix.java-version }}
+          path: gh-pages/reports/${{ inputs.agent-ref }}/${{ github.run_id }}/unit-tests-results-java-${{ matrix.java-version }}
+
+      - name: Set up Python
+        if: ${{ failure() }}
+        uses: actions/setup-python@v3
+
+      #This step executes a Python script that reads and writes html.
+      #The script requires the beautifulsoup dependency, which parses html.
+      #For details, view the utils/README.md on the gh-pages branch.
+      - name: Generate Test Reports page
+        if: ${{ failure() }}
+        run: |
+          pip install beautifulsoup4
+          python gh-pages/utils/writeTestsFailuresPage.py gh-pages/reports/${{ inputs.agent-ref }}/${{ github.run_id }}/unit-tests-results-java-${{ matrix.java-version }}
+
+      - name: Commit and push test reports to gh-pages
```

Comment:
The four java versions commit and push the reports to gh-pages in parallel. Currently, I don't have a retry in this step- it has not caused problems so far, as the four jobs are unlikely to finish at the exact same time.